src/toc_tree.tsx

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import * as React from 'react';
+import { Widget } from '@phosphor/widgets';
+import { IHeading } from './utils/headings';
+import { Registry } from './registry';
+import { TOCItem } from './toc_item';
+
+/**
+ * Interface describing component properties.
+ *
+ * @private
+ */
+interface IProperties extends React.Props<TOCTree> {
+  /**
+   * Display title.
+   */
+  title: string;
+
+  /**
+   * List of headings to render.
+   */
+  toc: IHeading[];
+
+  /**
+   * Toolbar.
+   */
+  toolbar: any;
```

Comment:
Memo to myself: we should tighten up the typings here.

Reply:
Yeah, was not sure the original intent, so left as is.